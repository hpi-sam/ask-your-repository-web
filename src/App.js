// @flow
import React from 'react';
import './App.scss';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          The Deployment works!!!
        </p>
      </header>
    </div>
  );
}

export default App;
