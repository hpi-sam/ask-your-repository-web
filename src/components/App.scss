@import '../style/breakpoints.scss';

.App {
  height: 100%;

  &__inner {
    max-width: var(--max-width);
    height: calc(100% - var(--navbar-height));

    margin: 0 auto;
    padding: 0 var(--padding-sides) 0px var(--padding-sides);
    padding-top: var(--navbar-height);

    display: flex;
    align-items: flex-start;
    justify-content: center;

    @media (min-width: $xs-to-sm) {
      transition: margin-left var(--sidebar-animation-duration);
      &--with-sidebar { margin-left: var(--team-sidebar-width); }
    }

    &--without-navbar {
      height: 100%;
      padding: auto;
    }
  }
}
