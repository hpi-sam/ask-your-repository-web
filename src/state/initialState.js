// @flow

// This file is only for state related testing

import type { AppState } from './AppState';
import { initialState as presentation } from './presentation/presentation.reducer';
import { initialState as teamSidebar } from './team_sidebar/teamSidebar.reducer';
import { initialState as presentationMode } from './presentation_mode/presentationMode.reducer';
import { initialState as activeTeam } from './active_team/activeTeam.reducer';
import { initialState as auth } from './auth/auth.reducer';

const initialState: AppState = {
  flash: { messages: [] },
  presentation,
  activeTeam,
  teamSidebar,
  presentationMode,
  auth,
};

export default initialState;
