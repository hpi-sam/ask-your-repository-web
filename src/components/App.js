// @flow
import React from 'react';
import { Switch, Route, Redirect } from 'react-router';
import FlashMessages from './utility/flash/FlashMessages';
import ImagesIndex from './images/ImagesIndex';
import FileUpload from './upload/FileUpload';
import Tagging from './tagging/Tagging';
import './App.scss';

function App() {
  return (
    <div className="App">
      <FlashMessages />
      <Switch>
        <Redirect exact from="/" to="/upload" />
        <Route path="/upload" component={FileUpload} />
        <Route path="/tagging" component={Tagging} />
        <Route path="/images" component={ImagesIndex} />
      </Switch>
    </div>
  );
}

export default App;
