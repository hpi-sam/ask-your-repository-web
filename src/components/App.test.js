// @flow
import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import initialState from '../state/initialState';
import App from './App';

const mockStore = configureMockStore();

describe('<App />', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount((
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    )).find('App');
  });

  it('renders correctly', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
