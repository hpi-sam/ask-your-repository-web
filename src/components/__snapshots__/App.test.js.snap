// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <Connect(FlashMessages) />
  <Switch>
    <Redirect
      exact={true}
      from="/"
      push={false}
      to="/upload"
    />
    <Route
      component={[Function]}
      path="/upload"
    />
    <Route
      component={[Function]}
      path="/tagging"
    />
    <Route
      component={[Function]}
      path="/images"
    />
  </Switch>
</div>
`;
